lightbeam/test.py

```diff
@@ -0,0 +1,214 @@
+        self.lightbeam.validator.validate()
+        # TODO: Uncomment the following once stuctured logging for validate is implemented
+        # expected_validate_file = os.path.join( tests_dir, "expected", "validate.json" )
+        # if not self.log_files_are_identical(validate_file, expected_validate_file):
+        #     self.logger.critical("`lightbeam validate` did not return the expected output")
+        
+        # * Test `lightbeam send`:
+        self.logger.info("testing `lighbeam send`...")
+        pre_send_count_file = os.path.join( tests_dir, "output", "send-pre-count.tsv" )
+        self.lightbeam.results_file = pre_send_count_file
+        self.lightbeam.selector = "schools,educationServiceCenters"
+        self.lightbeam.api.prepare()
+        self.lightbeam.counter.count()
+        pre_count = self.load_count_file(pre_send_count_file)
+        
+        send_file = os.path.join( tests_dir, "output", "send.json" )
+        self.lightbeam.results_file = send_file
+        self.lightbeam.api.prepare()
+        self.lightbeam.sender.send()
+        expected_send_file = os.path.join( tests_dir, "expected", "send.json" )
+        if not self.log_files_are_identical(send_file, expected_send_file):
+            self.logger.critical("`lightbeam send` did not return the expected output")
+        
+        post_send_count_file = os.path.join( tests_dir, "output", "send-post-count.tsv" )
+        self.lightbeam.results_file = post_send_count_file
+        self.lightbeam.selector = "schools,educationServiceCenters"
+        self.lightbeam.api.prepare()
+        self.lightbeam.counter.count()
+        post_count = self.load_count_file(post_send_count_file)
+        
+        for k in set(pre_count.keys()).union(post_count.keys()):
+            if pre_count.get(k, 0) + 1 != post_count.get(k, 0):
+                self.logger.critical(f"`lightbeam send` did not send the expected number of records for {k}")
+
+        # * Test `lightbeam delete`:
+        self.logger.info("testing `lighbeam delete`...")
+        self.lightbeam.results_file = ""
+        self.lightbeam.api.prepare()
+        self.lightbeam.deleter.delete()
+
+        post_delete_count_file = os.path.join( tests_dir, "output", "delete-post-count.tsv" )
+        self.lightbeam.results_file = post_delete_count_file
+        self.lightbeam.selector = "schools,educationServiceCenters"
+        self.lightbeam.api.prepare()
+        self.lightbeam.counter.count()
+        if not self.count_files_are_identical(pre_send_count_file, post_delete_count_file):
+            self.logger.critical("`lightbeam delete` did not affect the expected number of records")
+        
+        # * Test `lightbeam truncate`:
+        self.logger.info("testing `lighbeam truncate`...")
+        pre_truncate_count_file = os.path.join( tests_dir, "output", "truncate-pre-count.tsv" )
+        self.lightbeam.results_file = pre_truncate_count_file
+        self.lightbeam.selector = "disciplineActions"
```

Comment:
I use `disciplineActions` to test `truncate` because (1) there are a few `disciplineActions` in the Grand Bend dataset and (2) it's a "leaf" resource in the dependency tree, so deleting records won't cause dependency errors.